src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState(Status.All);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.warn(error);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    const lowerQuery = query.toLowerCase();
+
+    return todos.filter(todo => {
+      switch (status) {
+        case Status.Completed:
+          return todo.completed;
+
+        case Status.Active:
+          return !todo.completed;
+
+        default:
+          return true;
+      }
+    })
+      .filter(todo => todo.title.toLocaleLowerCase().includes(lowerQuery));
+  }, [todos, query, status]);
+
```

Comment:
Array is relatively small, and the filtering operation is fast, you might not see a significant performance difference by using useMemo. 
In such cases, you can choose to omit it for the sake of simplicity.